src/rustdoc-internals/htmldocck.md

```diff
@@ -0,0 +1,101 @@
+# `rustdoc` tests
+
+This page is specifically about the `rustdoc` test suite, for other test suites used for testing rustdoc, see [Rustdoc § Tests](../rustdoc.md#tests).
+
+`htmldocck.py` is a custom checker script that uses [XPath] to verify the HTML output of rustdoc.
+
+[XPath]: https://en.wikipedia.org/wiki/XPath
+
+## Directives
+Directives to htmldocck are similar to those given to `compiletest` in that they take the form of `//@` comments.
+
+All `PATH`s in directives are relative to the the rustdoc output directory (`build/TARGET/test/rustdoc/TESTNAME`),
+so it is conventional to use a `#![crate_name = "foo"]` attribute to avoid writing paths.
+To avoid repetion, `-` can be used in any `PATH` argument to re-use the previous `PATH` argument.
+
+All arguments take the form of quoted strings,
```

Comment:
You might want to mention that both single quotes as well as double quotes are supported.